rust/operator-binary/src/controller.rs

```diff
@@ -0,0 +1,890 @@
+            }
+            PropertyNameKind::File(file_name) if file_name == PASSWORD_AUTHENTICATOR_PROPERTIES => {
+                let pw_properties =
+                    product_config::writer::to_java_properties_string(transformed_config.iter())
+                        .context(PropertiesWriteError)?;
+                cm_conf_data.insert(file_name.to_string(), pw_properties);
+            }
+            PropertyNameKind::File(file_name) if file_name == PASSWORD_DB => {
+                // make sure password db is created to fill it via container command scripts
+                cm_conf_data.insert(file_name.to_string(), "".to_string());
+            }
+            PropertyNameKind::File(file_name) if file_name == JVM_CONFIG => {
+                jvm_config.push_str(&format!("\n-javaagent:/stackable/jmx/jmx_prometheus_javaagent-0.16.1.jar={}:/stackable/jmx/config.yaml", METRICS_PORT));
+            }
+            _ => {}
+        }
+    }
+
+    if let Some(opa) = opa_connect {
+        let package = match trino.spec.authorization.as_ref() {
+            Some(auth) => auth.package.clone(),
+            None => {
+                println!("No package specified in 'authorization'. Defaulting to 'trino'.");
+                "trino".to_string()
+            }
+        };
+
+        let mut opa_config = BTreeMap::new();
+
+        opa_config.insert(
+            "access-control.name".to_string(),
+            Some("tech.stackable.trino.opa.OpaAuthorizer".to_string()),
+        );
+        opa_config.insert(
+            "opa.policy.uri".to_string(),
+            Some(format!("{}v1/data/{}/", opa, package)),
+        );
+
+        let config_properties =
+            product_config::writer::to_java_properties_string(opa_config.iter())
+                .context(PropertiesWriteError)?;
+
+        cm_conf_data.insert("access-control.properties".to_string(), config_properties);
+    }
+
+    cm_conf_data.insert(JVM_CONFIG.to_string(), jvm_config.to_string());
+
+    ConfigMapBuilder::new()
+        .metadata(
+            ObjectMetaBuilder::new()
+                .name_and_namespace(trino)
+                .name(rolegroup_ref.object_name())
+                .ownerreference_from_resource(trino, None, Some(true))
+                .context(ObjectMissingMetadataForOwnerRef)?
+                .with_recommended_labels(
+                    trino,
+                    APP_NAME,
+                    trino_version(trino)?,
+                    &rolegroup_ref.role,
+                    &rolegroup_ref.role_group,
+                )
+                .build(),
+        )
+        .data(cm_conf_data)
+        .build()
+        .with_context(|| BuildRoleGroupConfig {
+            rolegroup: rolegroup_ref.clone(),
+        })
+}
+
+/// The rolegroup catalog [`ConfigMap`] configures the rolegroup catalog based on the configuration
+/// given by the administrator
+fn build_rolegroup_catalog_config_map(
```

Comment:
Is there a reason that this is separate from the other CM?

Reply:
The hive properties belong into `conf/catalog/hive.properties`, where `conf` is the first configmap mount and `conf/catalog` the second. Is there a better way?

Reply:
You can alter the mapping between keys and file paths using `Pod.spec.volumes[_].configMap.items`, but on second thought this kind of makes sense since we probably don't want to bake the exact catalogs into the pod definitions.